ptvsd/attach_server.py

```diff
@@ -88,8 +88,12 @@ def break_into_debugger():
     """If a remote debugger is attached, pauses execution of all threads,
     and breaks into the debugger with current thread as active.
     """
-    if not debugger_attached.isSet() or not _enabled:
-        return
+    if is_launch():
+        if not is_attached():
+            return
+    else:
+        if not is_attached() or not _enabled:
+            return
     import sys
```

Comment:
You can movie is_attached out and reduce the nesting of the conditions by one level.

Reply:
I have another PR for Server attach fix where this is addressed where this is addressed,